examples/jetstream/limits-stream/java/Main.java

```diff
@@ -0,0 +1,135 @@
+public class Main {
+    public static void main(String[] args) {
+        String natsURL = System.getenv("NATS_URL");
+        if (natsURL == null) {
+            natsURL = "nats://127.0.0.1:4222";
+        }
+
+        // Initialize a connection to the server. The connection is AutoCloseable
+        // on exit.
+        try (Connection nc = Nats.connect(natsURL)) {
+
+            // Access `JetStream` and `JetStreamManagement` which provide methods to create
+            // streams and consumers as well as convenience methods for publishing
+            // to streams and consuming messages from the streams.
+            JetStream js = nc.jetStream();
+            JetStreamManagement jsm = nc.jetStreamManagement();
+
+            // We will declare the initial stream configuration by specifying
+            // the name and subjects. Stream names are commonly uppercased to
+            // visually differentiate them from subjects, but this is not required.
+            // A stream can bind one or more subjects which almost always include
+            // wildcards. In addition, no two streams can have overlapping subjects
+            // otherwise the primary messages would be persisted twice. There
+            // are options to replicate messages in various ways, but that will
+            // be explained in later examples.
+            //
+            // JetStream provides both file and in-memory storage options. For
+            // durability of the stream data, file storage must be chosen to
+            // survive crashes and restarts. This is the default for the stream,
+            // but we can still set it explicitly.
+            String streamName = "EVENTS";
+            StreamConfiguration.Builder streamConfigBuilder = StreamConfiguration.builder()
+                    .name(streamName)
+                    .subjects("events.>")
+                    .storageType(StorageType.File);
+
+            // Finally, let's add/create the stream with the default (no) limits.
+            StreamInfo stream = jsm.addStream(streamConfigBuilder.build());
+            System.out.println("Created the stream.");
+
+            // Let's publish a few messages which are received by the stream since
+            // they match the subject bound to the stream. The `js.publish` method
+            // is a convenience for sending a `nc.request` and waiting for the
+            // acknowledgement.
+            js.publish("events.page_loaded", null);
+            js.publish("events.mouse_clicked", null);
+            js.publish("events.mouse_clicked", null);
+            js.publish("events.page_loaded", null);
+            js.publish("events.mouse_clicked", null);
+            js.publish("events.input_focused", null);
+            System.out.println("Published 6 messages.");
+
+            // There is also an async form in which the client batches the
+            // messages to the server and then asynchronously receives
+            // the acknowledgements.
```

Comment:
I don't know what go has, but java does not batch anything. Each publish async just publishes immediately and makes a future waiting for the pub ack instead of blocking until the pub ack comes back.

Reply:
Quickly glancing over the Go code it does seem the same implementation as in Java, not necessarily batch but allow multiple messages to be sent individually and collecting the acks asynchronously.

Have rephrased it to:
```java
// There is also an async form in which the client can send multiple
// messages to the server and then asynchronously receives the acknowledgements.
```

Only changed it in the Java example for now, not sure if we'd like to update the comment for the Go example as well?

Reply:
leave the others for now but maybe ask Piotr/Wally to review